scrapy/core/downloader/handlers/http11.py

```diff
@@ -454,6 +473,12 @@ def dataReceived(self, bodyBytes):
         self._bodybuf.write(bodyBytes)
         self._bytes_received += len(bodyBytes)
 
+        self._crawler.signals.send_catch_log(
+            signal=signals.bytes_received,
+            data=bodyBytes,
+            request=self._request,
+        )
```

Comment:
When the signal is not assigned to any handler, does this have any noticeable impact on performance?

Reply:
Using https://github.com/scrapy/scrapy-bench:

`scrapy-bench --n-runs=10 --book_url=http://localhost:8000 bookworm`

scrapy @ 454de31d20d1d7676c8db0efaf46da4c671f81e6 (`bytes_received_signal` branch)
```
The results of the benchmark are (all speeds in items/sec) : 
Test = 'Book Spider' Iterations = '10'
Mean : 62.16695929434989 Median : 62.43078124454085 Std Dev : 1.1148966043160284
```

scrapy @ 07b8cd28aa84fb322a072467b49e2557d4aa3881 (`master` branch)
```
The results of the benchmark are (all speeds in items/sec) : 
Test = 'Book Spider' Iterations = '10'
Mean : 60.77521979854436 Median : 60.963007266556204 Std Dev : 1.6349215293441601
```